### PR TITLE
Added support for bool type to the cpolicy package issue #734 

### PR DESCRIPTION
**Pull request by sandlbn**

### CODE REVIEW

File: control/plugin/cpolicy/bool.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2015 Intel Corporation
```
Comment (IzabellaRaulin) [inline control/plugin/cpolicy/bool.go]: 2015 -> 2016

File: control/plugin/cpolicy/bool_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			Convey("errors with non-string config value", func() {
+				r, e := NewBoolRule("thekey", true, true)
+				So(r.Default().Type(), ShouldEqual, "bool")
+				So(r.Default().(*ctypes.ConfigValueBool).Value, ShouldEqual, true)
+				So(e, ShouldBeNil)
+
+				v := ctypes.ConfigValueInt{Value: 1}
+
+				e = r.Validate(v)
+				So(e, ShouldResemble, errors.New("type mismatch (thekey wanted type 'bool' but provided type 'integer')"))
```
Comment (IzabellaRaulin) [inline control/plugin/cpolicy/bool_test.go]: space missing in "thekey"

File: control/plugin/cpolicy/bool.go (hunk 1 above)
Comment (sandlbn) [inline control/plugin/cpolicy/bool.go]: Thanks, I'll rebase in a moment.

File: control/plugin/cpolicy/bool_test.go (hunk 2 above)
Comment (sandlbn) [inline control/plugin/cpolicy/bool_test.go]: Yes, that was done intentionally.

**Comment by sandlbn** [comment]
Any new comments ? I need this merged, to push code for 2 plugins.

File: control/plugin/cpolicy/bool.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+// NewBoolRule Returns a new bool-typed rule. Arguments are key(string), required(bool), default(bool).
+func NewBoolRule(key string, req bool, opts ...bool) (*BoolRule, error) {
+	// Return error if key is empty
+	if key == "" {
+		return nil, EmptyKeyError
+	}
+
+	options := make([]*bool, 1)
```
Comment (danielscottt) [inline control/plugin/cpolicy/bool.go]: It took me a minute to understand what `options` was for.

You can use <code>opts[_n_]</code>, so maybe after creating your instance below, you could instead do something like this:

```go
b := &BoolRule {
        key:      key,
        required: req,
}

// test whether or not a default was given.
if len(opts) != 0 {
        b.default_ = opts[0]
}
return b, nil
```

Comment (danielscottt) [inline control/plugin/cpolicy/bool.go]: Ping @sandlbn -- Do you have any opinion about this comment?  If you disagree, that's fine, I'd just like to get this merged for you!

Comment (sandlbn) [inline control/plugin/cpolicy/bool.go]: @danielscottt sorry to much meetings today, I'll update my PR.

**Comment by sandlbn** [comment]
@danielscottt ok done.

**Comment by danielscottt** [comment]
LGTM.  Merging.

**Pull request merged**